File: tasks/implementations/node/tasks/image_resizing.js
const fs = require('fs');
module.exports = function(size, fixturesRoot) {
  const tokens = fs.readFileSync(`${fixturesRoot}/generated/image/${size}.ppm`, 'utf8').trim().split(/\s+/);
  const width = Number(tokens[1]);
  const height = Number(tokens[2]);
  const data = tokens.slice(4).map(Number);
  let checksum = 0;
  for (let y = 0; y < height / 2; y++) {
    for (let x = 0; x < width / 2; x++) {
      const idx = ((y * 2) * width + (x * 2)) * 3;
      checksum += data[idx] + data[idx + 1] + data[idx + 2];
    }
  }
  console.log(checksum);
};

File: tasks/implementations/rust/tasks/image_resizing.rs
use std::fs;

pub fn run(size: &str, fixtures_root: &str) {
    let text = fs::read_to_string(format!("{}/generated/image/{}.ppm", fixtures_root, size)).unwrap();
    let tokens: Vec<i64> = text.split_whitespace().skip(1).map(|x| x.parse::<i64>().unwrap_or(0)).collect();
    let width = tokens[0] as usize;
    let height = tokens[1] as usize;
    let data = &tokens[3..];
    let mut checksum = 0i64;
    for y in 0..(height / 2) {
        for x in 0..(width / 2) {
            let idx = ((y * 2) * width + (x * 2)) * 3;
            checksum += data[idx] + data[idx + 1] + data[idx + 2];
        }
    }
    println!("{}", checksum);
}
